src/solver/simulation/timeseries-numbers.cpp

```diff
@@ -120,155 +81,601 @@ namespace Antares
 {
 namespace Solver
 {
-template<enum Data::TimeSeries TimeSeriesT>
-static unsigned int CheckMatricesWidth(const Data::Study& study)
+static bool GenerateDeratedMode(Data::Study& study)
+{
+    logs.info() << "  :: using the `derated` mode";
+    if (study.parameters.useCustomScenario)
+        logs.warning() << "The derated mode is enabled. The custom building mode will be ignored";
+
+    study.areas.each([&](Data::Area& area) {
+        area.load.series->timeseriesNumbers.zero();
+        area.solar.series->timeseriesNumbers.zero();
+        area.wind.series->timeseriesNumbers.zero();
+        area.hydro.series->timeseriesNumbers.zero();
+
+        for (uint i = 0; i != area.thermal.clusterCount(); ++i)
+        {
+            auto& cluster = *(area.thermal.clusters[i]);
+            cluster.series->timeseriesNumbers.zero();
+        }
+    });
+
+    return true;
+}
+
+
+class areaTSNumbersListRetriever
+{
+public:
+    areaTSNumbersListRetriever(Data::Study& study) : study_(study) {}
+    virtual std::vector<uint> getAreaTimeSeriesNumber(const Data::Area& area) = 0;
+    virtual uint getGeneratedTimeSeriesNumber() = 0;
+protected:
+    Data::Study& study_;
+};
+
+class areaLoadTSNumbersListRetriever : public areaTSNumbersListRetriever
+{
```

Comment:
Keep the class interface, but use templates to name derived classes ?

Something like that
```cpp
class Interface {
 virtual void f() = 0;
};

template<int mode>
struct Derived;

template<>
struct Derived<timeSeriesThermal> : public Interface {
// ...
};
```

Reply:
it seems excessive to use the TS kind as a template parameter to make Derived instance more understandable.
Indeed, the template parameter is not meant to be used inside the class. 
After we talked about it, we decided to rename **areaTSNumbersListRetriever**'s children in such a way that the kind of TS is highlighted.